File: android/app/src/main/kotlin/io/rebble/cobble/datasources/WatchMetadataStore.kt
package io.rebble.cobble.datasources

import io.rebble.libpebblecommon.packets.WatchVersion
import kotlinx.coroutines.flow.MutableStateFlow
import java.util.UUID
import javax.inject.Inject
import javax.inject.Singleton

//TODO: Consolidate this with the shared ConnectionStateManager
@Singleton
class WatchMetadataStore @Inject constructor() {
    val lastConnectedWatchMetadata = MutableStateFlow<WatchVersion.WatchVersionResponse?>(null)
    val lastConnectedWatchModel = MutableStateFlow<Int?>(null)
    val currentActiveApp = MutableStateFlow<UUID?>(null)
}

File: ios/Runner/datasources/WatchMetadataStore.swift
//
//  WatchMetadataStore.swift
//  Runner
//
//  Created by crc32 on 27/02/2022.
//

import Foundation
import libpebblecommon

class WatchMetadataStore {
    static var shared: WatchMetadataStore!
    var lastConnectedWatchMetadata: WatchVersion.WatchVersionResponse? = nil
    var lastConnectedWatchModel: Int? = nil
}
